src/services/Fedora.ts

```diff
@@ -1,26 +1,75 @@
+import { IncomingMessage } from "http";
 const http = require("needle");
 
+interface NeedleResponse extends IncomingMessage {
+    body: any;
+    raw: Buffer;
+    bytes: number;
+}
+
+interface Attributes {
+    [key: string]: string;
+}
+
+interface JSON {
+    [key: string]: string | JSON;
+}
+
+interface DC {
+    name: string;
+    value: string;
+    attributes: Attributes;
+    children: Array<DC>;
+}
+
 class Fedora {
     baseUrl: string;
 
     constructor() {
+        // TODO: Config
         this.baseUrl = "http://vua6743.villanova.edu:8089/rest";
     }
 
-    async getDC(pid) {
-        let options = {
+    /**
+     * Make authenticated request to Fedora
+     */
+    protected _request(
+        method: string = "get",
+        _path: string = "/",
+        data: object = null
+    ): Promise<NeedleResponse> {
+        let path = _path[0] == "/" ? _path.slice(1) : _path;
+        let url = this.baseUrl + "/" + path;
+
+        // TODO: Config
+        const headers = {
             username: "fedoraAdmin", // Basic Auth
-            password: "fedoraAdmin"
+            password: "fedoraAdmin",
         };
+
+        return http("get", url, null, headers);
+    }
+
+    /**
+     * Get datastream from Fedora
+     */
```

Comment:
Might be a bit much for this PR but what do you think?

Reply:
Seems along the lines of what I was thinking of! It will take more time and experimentation to identify whether or not the data types work exactly as they are, but I'm happy with picking an arbitrary starting point and evolving as needs demand.